### ISSUE TITLE
Please update firebase-messaging dependency

### ISSUE THREAD

**Issue by darnmason** [urbanairship/android-library#101]
Here's my dependency tree, Google is complaining about firebase-iid version compatibility as I'm using the latest versions of firebase-core and firebase-perf. Trying to find a combination of compatible versions is next to impossible when you throw Google Ads into the mix. Can you update to the latest version of firebase-messaging (17.3.4) which will resolve this.

For now I've excluded it from UA and included directly in my dependencies using latest version and _hoping_ it's not breaking anything.

Cheers

```+--- com.urbanairship.android:urbanairship-fcm:9.5.6
|    +--- com.google.firebase:firebase-messaging:17.0.0
|    |    +--- com.google.android.gms:play-services-basement:15.0.1 -> 16.0.1 (*)
|    |    +--- com.google.android.gms:play-services-tasks:15.0.1 -> 16.0.1 (*)
|    |    +--- com.google.firebase:firebase-common:16.0.0 -> 16.0.3 (*)
|    |    +--- com.google.firebase:firebase-iid:[16.0.0] -> 17.0.3 (*)
|    |    \--- com.google.firebase:firebase-measurement-connector:16.0.0 -> 17.0.1 (*)
|    +--- com.google.android.gms:play-services-base:15.0.1 -> 16.0.1 (*)
|    \--- com.urbanairship.android:urbanairship-core:9.5.6
|         +--- com.android.support:support-v4:27.1.1 (*)
|         \--- com.android.support:support-annotations:27.1.1
```

**Comment by rlepinski** [comment]
@darnmason Your current workaround should fine. We will take a look at making it less awful for people with firebase dependency issues. It use to be you could just specify a newer FCM version in your app to have our SDK use it, but now that throws alls sorts of issues due to the strict version plugin. Ill take a look at ranged versions, excluding the dependency all together, or simply just updated it for the next release. The release probably wont make it out til Wednesday or Friday, but your current workaround should be fine.

Could you provide me with your firebase and google services modules and versions you are using so I can reproduce your exact failure? Thanks!

**Comment by darnmason** [comment]
Yeah Firebase is a pain these days.

Cheers, I've got:
```
com.google.firebase:firebase-core:16.0.6
com.google.firebase:firebase-perf:16.2.2
com.google.android.gms:play-services-ads:17.1.2
com.urbanairship.android:urbanairship-fcm:9.5.6
```

**Comment by rlepinski** [comment]
Released 9.6.0 that uses firebase-messaging 17.3.4.
